### PR TITLE
ISPN-10367 Possible lock lost if a transaction reaches timeout

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-10367

### CODE REVIEW

File: core/src/main/java/org/infinispan/commands/remote/CheckTransactionRpcCommand.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+   public CheckTransactionRpcCommand(ByteString cacheName) {
+      this.cacheName = cacheName;
+   }
+
+   public static ResponseCollector<Collection<GlobalTransaction>> responseCollector() {
+      return INSTANCE;
+   }
+
+   @Override
+   public void init(ComponentRegistry componentRegistry, boolean isRemote) {
```
[the remark this answers is not in the document]
Comment (pruivo) [reply core/src/main/java/org/infinispan/commands/remote/CheckTransactionRpcCommand.java]: no, I don't need it.

**Comment by pruivo** [comment]
rebased!

**Comment by ryanemerson** [comment]
@pruivo Test Result (1,252 failures / +1245)

**Comment by pruivo** [comment]
only thread leaks!? they don't seem related.
rebased on top of master... lets wait for a new CI run.

**Comment by pruivo** [comment]
CI looks fine now :)

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**